### ISSUE TITLE
Provide design for links that will open in a new tab/window

### ISSUE THREAD

**Issue by mollykreis** [ni/nimble#1412]
## 📌 Request Background

According to [WCAG guidelines](https://www.w3.org/TR/WCAG20-TECHS/G201.html), users should be clearly informed if a link they are about to click on will open in a new tab/window. Today in nimble, we have no design guidelines on how this should be communicated to a user. A few options could be:
- Add text to the link that indicates what will happen (e.g. "Preview in a new tab" or "View in a new window" rather than simply "Preview" or "View")
- Add an icon that indicates intent

## 🎯 Core Requirements

How should a link that will always open in a new tab/window be represented to a user? In nimble, links can be represented in a few different ways, so we should have guidance and/or designs for all possible links:
- [nimble-anchor](https://nimble.ni.dev/storybook/?path=/docs/components-anchor--docs)
- [nimble-anchor-menu-item](https://nimble.ni.dev/storybook/?path=/docs/components-menu--docs#anchor-menu-item)
- [nimble-anchor-tree-item](https://nimble.ni.dev/storybook/?path=/docs/components-tree-view--docs#anchor-tree-item)

## 🍆 Out of scope

- [nimble-anchor-button](https://nimble.ni.dev/storybook/?path=/docs/components-anchor-button--docs) ❌
- [nimble-anchor-tabs](https://nimble.ni.dev/storybook/?path=/docs/components-anchor-tabs--docs) ❌

## 🥅 Acceptance Criteria

- [x] Visual Design Spec submitted (link)
- [x] Visual Design Spec reviewed by design team
- [x] [Visual Design Spec](https://www.figma.com/file/PO9mFOu5BCl8aJvFchEeuN/Nimble_Components?type=design&node-id=1295-76706&mode=dev) reviewed by dev team

**Comment by mollykreis** [comment]
This issue should be evaluated alongside #1331

**Comment by leslieab** [comment]
From a Figma video I was watching.


![Image](https://github.com/ni/nimble/assets/28602266/9fc15f5f-bab6-452a-b1e5-0244308873cf)

**Comment by fredvisser** [comment]
@NIbokeefe - Could we get a nimble-icon version of [arrow-up-right-from-square](https://fontawesome.com/icons/arrow-up-right-from-square?f=classic&s=solid)?

![Image](https://github.com/ni/nimble/assets/1458528/7827e004-f087-43c7-a0da-70b7895a4c58)

**Comment by jattasNI** [comment]
In our discussion today we briefly debated about showing this icon on other kinds of anchors: anchor buttons and anchor tabs. We decided to start with those being out of scope. It seems likely that opening an anchor tab in a new tab automatically is always a bad idea. And we couldn't think of high priority use cases for automatically opening a new tab when you click an anchor button.

**Comment by NIbokeefe** [comment]
@fredvisser We should review the usage guidelines to make sure all questions are answered before calling this done.

- Component assets update
- Usage guidelines updated

https://www.figma.com/file/PO9mFOu5BCl8aJvFchEeuN/Nimble_Components?type=design&node-id=1295%3A76706&mode=dev

**Comment by jattasNI** [comment]
@NIbokeefe the team reviewed these designs today. Looking good! Here are our questions:
1. do you have a design for the anchor tree item?
2. there is usage guidance that clients shouldn't use an icon-only anchor button for links that open in a new tab. That makes sense to us (there wouldn't be a good way to display both the button icon and the external link icon in a small square). Does that guidance also extend to "don't use an icon-only button for links generally (even if the link will open in the same tab)?

**Comment by NIbokeefe** [comment]
@jattasNI Thanks for reviewing that.

1.  I didn't draw a tree item scenario, but will add that.  It will be very similar to the menu execution.
2. I think that's correct, you wouldn't use an icon only button **alone** for links or anchor links in general, you would instead use an icon-text button.  However, I have seen Leslie and other UX people draw a link or copy icon-only-button in combination with a text input field.  I can't find the wireframe off hand, but had mapped out an execution like this example:
<img width="976" alt="Screen Shot 2024-02-27 at 1 48 54 PM" src="https://github.com/ni/nimble/assets/106594349/0373f988-4a2e-4225-bfaa-caf5e11e509d">
ution for it previously.

**Comment by rajsite** [comment]
@fredvisser do we have issues created for the open in new window icon support on the required elements? Wherever those are we should capture [this implementation guidance](https://dev.azure.com/ni/DevCentral/_git/Skyline/pullRequest/907533#1740172602).

**Comment by m-akinc** [comment]
> [@m-akinc](https://github.com/m-akinc) I see the triage tag was removed that I added for the last comment in the thread, but don't see any follow up discussion. What was the result of discussing in triage?

Sorry, didn't realize there was a specific reason for the triage tag. I'll add it back.
